### ISSUE TITLE
Fine-tuning on TABFACT

### ISSUE THREAD

**Issue by r08922129** [google-research/tapas#93]
Hello, I have a question.
I want to fine-tune tapas on TABFACT. After copying code on colab to local and run it for prediction. I think the following setting is correct.
```
python3 tapas/run_task_main.py \
  --task="TABFACT" \
  --output_dir="results" \
  --init_checkpoint="tapas_model/model.ckpt" \
  --bert_config_file="tapas_model/bert_config.json" \
  --mode="train" \
  --train_batch_size=6
```
However, the process finished with following message
```
Key output_bias/adam_m not found in checkpoint
	 [[node save/RestoreV2 (defined at /home/allen/anaconda3/envs/tabfact/lib/python3.7/site-packages/tensorflow_estimator/python/estimator/estimator.py:1514) ]]
```
Could anyone help? Many thanks.

**Comment by eisenjulian** [comment]
Hi @r08922129, thanks for reaching out. Is that message causing a failure? I am asking since depending of the checkpoint you are loading, it's expected the weights for the output layer will not be present, but that shouldn't create any problems. Can you double check if there are not other errors? Ideally if we can see a minimally reproducible example of the issue, in a colab for instance, we'll be able to help you much faster. Does the TabFact colab work for you?

**Comment by r08922129** [comment]
Thanks your reply. The colab you provide works in prediction.
I added the command mentioned above in the last cell for training.
Here is the code to reproduce this problem. [https://colab.research.google.com/drive/1a5L6rO5Df4zySkJAxvKQsM4DzqeEhrkg?usp=sharing](url)

**Comment by eisenjulian** [comment]
Hello @r08922129 thanks for sending the colab to reproduce the issue. I have found the problem, which is a bit tricky to explain. 

Basically there are two ways to supply a pre-trained model, one is through the init_checkpoint flag, and the other through the output_dir/model_dir flags, which points to the location used to save models, predictions and tensorboard summaries during training every X number of steps. Typically if you follow the instructions in the [README.md](https://github.com/google-research/tapas) we only use the init_checkpoint for the downloaded model, which is the correct way. The model_dir, in your case `result/tabfact/model`, should be always empty when you start.

The only exception to this rule is when you just want to download one of the pretrained models and predict with it, as we do in the colab, there you will see we copy the downloaded model into `result/tabfact/model` for prediction. This is because we implemented `run_task_main.py`to evaluate saved checkpoints during training, and then for the colab we had add this workaround. We will try to improve this use-case as well to prevent this confusion in the future.

Concretely, the problem here is you are mixing the two use-cases. The first part copied the code into `result/tabfact/model` but the training part expects it to be empty. If you try to delete the folder for example, or you change the output_dir/model_dir argument in run_task_main to point to a different location, the error goes away. Note you also don't have any training data in your colab, so the model won't train either.

Let me know if this is clear or there are any follow up questions.
